### PR TITLE
Terminal text cols adjusts to terminal container. 

### PR DESCRIPTION
**Pull request by Ozy-Viking**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/dockge/blob/master/CONTRIBUTING.md

Tick the checkbox if you understand [x]: 
- [x] I have read and understand the pull request rules.

# Description

Fixes #(223)

## Type of change

Please delete any options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- User interface (UI)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [x] My changes generate no new warnings
- [x] ~~My code needed automated testing. I have added them (this is optional task)~~

### Before Changes
image

### After Changes
image

### CODE REVIEW

**Comment by louislam** [comment]
The real terminal is in the backend using "node-pty".

It needs to update the the column in the backend too.

**Comment by Ozy-Viking** [comment]
I will have a look into that.

**Comment by Ozy-Viking** [comment]
I have added:

- Event listener when window resizes.
- Removing event listener.
- Updating backend where there was already a TODO item.

**Comment by louislam** [comment]
Thanks, but I believe that a lot of code changes are just format changes, not logic changes, which makes it harder for me to review your code. Please keep them unchanged. And should use project's eslint only, these commands may help:

```
pnpm run lint
pnpm run fmt
pnpm run check-ts
```

**Comment by Ozy-Viking** [comment]
I went back through using GitHub diff split view to remove any unnecessary changes, sorry about that.

File: package.json [hunk 1]
```diff
@@ -71,8 +69,10 @@
         "@typescript-eslint/eslint-plugin": "~6.8.0",
         "@typescript-eslint/parser": "~6.8.0",
         "@vitejs/plugin-vue": "~4.5.2",
+        "@xterm/addon-fit": "0.9.0-beta.1",
```
Comment (louislam) [inline package.json:72]: Is there any reason to use the specific beta version?

Comment (Ozy-Viking) [reply package.json:72]: No reason.

Comment (Ozy-Viking) [reply package.json:72]: I choose [@xterm/addon-fit](https://www.npmjs.com/package/@xterm/addon-fit) package over [xterm-addon-fit](https://www.npmjs.com/package/xterm-addon-fit) as it is the new addon format see [xterm README](https://github.com/xtermjs/xterm.js#addons) as of xterm v3.14.0.

Comment (louislam) [reply package.json:72]: I see, but since `0.9.0-beta.37` requires `@xterm/xterm` instead of `xterm`, I have to change this along with this pr.

**Pull request merged**